client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -321,6 +327,12 @@ public void noSuchSegment(NoSuchSegment noSuchSegment) {
         public void dataAppended(DataAppended dataAppended) {
             log.trace("Received ack: {}", dataAppended);
             long ackLevel = dataAppended.getEventNumber();
+            long previousAckLevel = dataAppended.getPreviousEventNumber();
+            try {
+                checkAckLevels(ackLevel, previousAckLevel);
+            } catch (Exception e) {
+                failConnection(e);
+            }
             ackUpTo(ackLevel);
```

Comment:
I don't know that we want to ack all the events if the verification has failed. It might be simpler to just let this happen via the re-connect. In which case these two lines should be moved into the try/catch

Reply:
fixed